### PR TITLE
Bug 1274546 - Initialize mozL10n on ToC startup

### PR DESCRIPTION
**Pull request by mancas**

### CODE REVIEW

File: add-on/panels/js/roomToc.jsx [hunk 1]
```diff
@@ -12,51 +13,80 @@ loop.roomToc = (function() {
...
+      var constants = results[requestIdx];
 
-    // XXX akita: Get loop constants
-    var roomStore = new loop.store.RoomStore(dispatcher, {
-      constants: { CONSTANT: false }
-    });
+      // Do the initial L10n setup, we do this before anything
+      // else to ensure the L10n environment is setup correctly.
+      var stringBundle = results[++requestIdx];
+      var locale = results[++requestIdx];
```
Comment (Mardak) [inline add-on/panels/js/roomToc.jsx]: We can use destructuring assignment and just do `let [constants, stringBundle, locale] = results` to not need all the `++requestIdx`.

File: add-on/panels/js/roomToc.jsx [hunk 2]
```diff
@@ -12,51 +13,80 @@ loop.roomToc = (function() {
...
-      constants: {},
-      useDataChannels: true,
-      dispatcher: dispatcher,
-      sdk: OT
-    });
+    var requests = [
+      ["GetAllConstants"],
+      ["GetAllStrings"],
+      ["GetLocale"],
+      ["GetLoopPref", "ot.guid"]
```
Comment (Mardak) [inline add-on/panels/js/roomToc.jsx]: Are we using this pref soon?

**Comment by mancas** [comment]
@Mardak comments addressed

File: add-on/panels/js/roomToc.jsx [hunk 3]
```diff
@@ -12,51 +13,71 @@ loop.roomToc = (function() {
...
-    loop.store.StoreMixin.register({
-      activeRoomStore: activeRoomStore,
-      roomStore: roomStore
-    });
+      var dispatcher = new loop.Dispatcher();
+      var sdkDriver = new loop.OTSdkDriver({
+        // For the standalone, always request data channels. If they aren't
+        // implemented on the client, there won't be a similar message to us, and
+        // we won't display the UI.
+        constants: {},
```
Comment (Mardak) [inline add-on/panels/js/roomToc.jsx]: Looks like most other places pass in constants, although I didn't really look at what it's being used for..

File: add-on/panels/js/roomToc.jsx [hunk 4]
```diff
@@ -12,51 +13,71 @@ loop.roomToc = (function() {
...
-                      isScreenShareActive={false} />, document.querySelector("#main"));
+      var roomStore = new loop.store.RoomStore(dispatcher, {
+        constants: constants
+      });
 
-    var locationData = sharedUtils.locationData();
-    var hash = locationData.hash.match(/#(.*)/);
+      loop.store.StoreMixin.register({
+        activeRoomStore: activeRoomStore,
+        roomStore: roomStore
```
Comment (Mardak) [inline add-on/panels/js/roomToc.jsx]: Fyi, with es6 you can leave off the value when the value is using a variable with the name of the key, e.g., `{ activeRoomStore, roomStore }` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#Property_definitions

**Pull request merged**